File: frontend/lib/modelos/encuesta.model.dart
import '../pantallas/my_home_page.dart';

class Encuesta {
  var id;
  var titulo;
  var descripcion;
  var estado;
  var fechaIni;
  var fechaFin;

  Encuesta({this.id, this.titulo, this.descripcion, this.estado, this.fechaIni, this.fechaFin});

  factory Encuesta.fromJson(Map<String, dynamic> json){
    return Encuesta(
      id: json['_id'],
      titulo: json['titulo'],
      descripcion: json['descripcion'],
      estado: json['estado'],
      fechaIni: json['fecha_ini'],
      fechaFin: json['fecha_fin']
    );
  }
}

File: backend/src/models/encuesta.model.js
const { Schema, model } = require('mongoose');
const encuestaSchema = new Schema({
    titulo: String,
    descripcion: String,
    estado: Number,
    fecha_ini: Date,
    fecha_fin: Date,
        Secciones: [{
            titulo: String,
            descripcion: String,
        TipoPregunta: [{
            nombre: String,
        pregunta: [{
            titulo: String,
        respuesta: [{
            trespuesta: String,
            }],
          }],
        }],
    }],
})

module.exports = model('Encuesta', encuestaSchema);